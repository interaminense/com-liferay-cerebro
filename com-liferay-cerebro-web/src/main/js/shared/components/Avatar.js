import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

const COLORS = ['danger', 'default', 'info', 'primary', 'success', 'warning'];

const SIZES = ['sm', 'lg', 'xl', 'xxl'];

function getInitials(first, last) {
	let retVal = first.substring(0, 1);

	if (last) {
		retVal += last.substring(0, 1);
	}

	return retVal;
}

class Avatar extends Component {
	render() {
		const {colorId, firstName, lastName, size, portraitURL} = this.props;

		const classes = getCN(
			'avatar-root',
			'rounded-circle',
			'sticker',
			`sticker-${COLORS[colorId % COLORS.length]}`,
			{
				[`sticker-${size}`]: size
			}
		);

		return (
			<span class={classes}>
				{
					do {
						if (portraitURL) {
							<img
								alt="thumbnail"
								class="img-fluid"
								src={portraitURL}
							/>;
						} else {
							getInitials(firstName, lastName);
						}
					}
				}
			</span>
		);
	}
}

Avatar.PROPS = {
	colorId: Config.number().value(0),
	firstName: Config.string().value(''),
	lastName: Config.string(),
	portraitURL: Config.string(),
	size: Config.oneOf(SIZES)
};

Avatar.COLORS = COLORS;
Avatar.SIZES = SIZES;

export default Avatar;