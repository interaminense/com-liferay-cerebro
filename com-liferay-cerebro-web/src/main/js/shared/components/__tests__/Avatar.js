import Avatar from '../Avatar';

describe('Avatar', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Avatar();

		expect(component).toMatchSnapshot();
	});

	it('should render with a first name', () => {
		component = new Avatar({
			firstName: 'foo'
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with a first and last name', () => {
		component = new Avatar({
			firstName: 'foo',
			lastName: 'bar'
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with a specific size', () => {
		component = new Avatar({
			size: 'xl'
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with a specific color id', () => {
		component = new Avatar({
			colorId: 4
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with a portraitURL', () => {
		component = new Avatar({
			portraitURL: 'http://i.imgur.com/G5pfP.jpg'
		});

		expect(component).toMatchSnapshot();
	});
});