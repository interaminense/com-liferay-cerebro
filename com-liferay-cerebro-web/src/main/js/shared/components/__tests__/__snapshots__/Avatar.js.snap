// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render 1`] = `<span class="avatar-root rounded-circle sticker sticker-danger"></span>`;

exports[`Avatar should render with a first and last name 1`] = `<span class="avatar-root rounded-circle sticker sticker-danger">fb</span>`;

exports[`Avatar should render with a first name 1`] = `<span class="avatar-root rounded-circle sticker sticker-danger">f</span>`;

exports[`Avatar should render with a portraitURL 1`] = `
<span class="avatar-root rounded-circle sticker sticker-danger">
  <img src="http://i.imgur.com/G5pfP.jpg" alt="thumbnail" class="img-fluid">
</span>
`;

exports[`Avatar should render with a specific color id 1`] = `<span class="avatar-root rounded-circle sticker sticker-success"></span>`;

exports[`Avatar should render with a specific size 1`] = `<span class="avatar-root rounded-circle sticker sticker-danger sticker-xl"></span>`;
